### ISSUE TITLE
Support Ethereum Recoverable Signatures

### ISSUE THREAD

**Issue by gatoWololo** [boltlabs-inc/tss-ecdsa#508]
We will need to support Ethereum [recoverable signatures](https://ethereum.stackexchange.com/questions/64380/understanding-ethereum-signatures). Notes:
- How do we expose this functionality? As an extra parameter to our function? This probably won't work as the `Output` associated type of the `ProtocolParticipant` trait is still set to `tss_ecdsa::sign::Signature`. It would be easier to just modify the existing `SignParticipant::compute_output` method to return the recoverable signature, at the cost of no longer having the "regular" signature...

We should be able to compute this signature as follows:
> Actually it seems like we can [generate the recovery ID if we know the public key and message](https://docs.rs/k256/latest/k256/ecdsa/struct.RecoveryId.html#method.trial_recovery_from_msg). If the coordinator doesn't have the public key by default, we could send it back from the nodes or have the nodes compute the RID themselves and send the (signature, RID) pair.

I personally don't know how "trial recovery" works.

**Comment by gatoWololo** [comment]
Update on this ticket:
We believe using the `trial_recovery_from_msg` function is the right approach.

We have a few choices about how to expose this signature to the end user:
For example, we could change the `Output` type of the `ProtocolParticipant` implementation on `SignParticipant`:
```rust
impl ProtocolParticipant for SignParticipant {
    type Input = Input;
    type Output = Signature;
```
To return a `RecoverableSignature`. I think this would be the wrong approach.

Instead, we should extend the API of our `Signature` type to support a `recovery_id` function:
```rust
fn recovery_id(&self, message: &[u8], public_key: &VerifyingKey) -> RecoveryId
```
Using k256 [RecoveryId](https://docs.rs/k256/latest/k256/ecdsa/struct.RecoveryId.html). This would allow a user to "opt-in" to a recoverable signature.

**Comment by gatoWololo** [comment]
Closed by #512
